src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogProperties.java

```diff
@@ -291,7 +291,7 @@ private void createTabSearch(PreferencesDialog window) {
 
         window.addGroup(Resources.getMessage("DialogProperties.10")); //$NON-NLS-1$
 
-        window.addPreference(new PreferenceBoolean(Resources.getMessage("PropertyDialog.44")) { //$NON-NLS-1$
+        window.addPreference(new PreferenceBoolean(Resources.getMessage("PropertyDialog.44"), true, true) { //$NON-NLS-1$
```

Comment:
How did you come to the conclusion that "true" should be the default here?

Reply:
I think that it should be "false"...

Reply:
> How did you come to the conclusion that "true" should be the default here?

I called the function in swtprefernce example and got Yes:

         window.addPreference(new PreferenceBoolean("BooleanPreference with default") {
            protected Boolean getValue() { return true; }
            protected void setValue(Object t) { /*(Boolean)t*/}
        });       

![Uploading Screenshot from 2021-12-19 17-41-54.png…]()

Reply:
Hmm. I don't understand. In the screenshot from you code you can see, that swtpreferences doesn't offer the option to set the default for ""BooleanPreference with default" (blue arrow is grayed out). So you either 

(1) need to introduce a new constructor into swtpreferences to set hideLabel without setting a default (because currently there is no default set in ARX for the preferences under discussion). That would be the 100% downwards-compatible solution. Feel free to implement it (add new constructor to PreferenceBoolan and update swtprefereces again).

(2) Introduce a default. But it has to be the correct default. And this has nothing to do with swtpreferences or the example. It has to do with ARX. Every preference exposes one configuration option of ARX.

Let us take a look at this specific example:

```java
window.addPreference(new PreferenceBoolean(Resources.getMessage("PropertyDialog.44"), true, true) { //$NON-NLS-1$
	protected Boolean getValue() { return model.getInputConfig().isHeuristicForSampleBasedCriteria(); }
	protected void setValue(Object t) { model.getInputConfig().setHeuristicForSampleBasedCriteria((Boolean)t); }});
}
```

We can see that this exposes the option accessible via:

```java
model.getInputConfig().isHeuristicForSampleBasedCriteria()
```

So you have to jump into the function ```isHeuristicForSampleBasedCriteria```:

```java
public boolean isHeuristicForSampleBasedCriteria() {
	return config.isUseHeuristicSearchForSampleBasedCriteria();
}
```

And then you have to jump into ```isUseHeuristicSearchForSampleBasedCriteria```

```java
public boolean isUseHeuristicSearchForSampleBasedCriteria() {
        return heuristicSearchForSampleBasedCriteria;
}
```

And now you can take a look at the variable and see what its default is:

```java
private boolean heuristicSearchForSampleBasedCriteria = false;
```

-> The default should be "false".

Reply:
Thanks for the details. I missed the blue arrow, sorry. If you don't mind I will go for the first solution.

Reply:
Update: I modified [line 49](https://github.com/iaBIH/swtpreferences/blob/e5f2b398663f6e313973a3eb8f024d025633140d/src/main/de/linearbits/preferences/PreferenceBoolean.java#L49) in PreferenceBoolean.java and now the blue arrow is active

![Screenshot from 2021-12-19 20-10-23](https://user-images.githubusercontent.com/28802516/146687760-c44be41a-d0cc-4246-a55b-2fbb90fa276c.png)

Next todo:
1. Push the updated swtpreferences code.
2. Correct the code in ARX i.e. replace true with false. 

Please let me know if you agree.

Reply:
I don't understand what you mean by 2. Why "replace true with false". I think that the default parameter would need to be removed. Regarding 1. I would suggest  to not change line 49 but to add a new constructor ```public PreferenceBoolean(String label, Boolean hideLabel)```.

Reply:
PS: Apart from that, I also think that it would be fine to keep the defaults in the ARX dialog. But they should be the correct defaults.

Reply:
>  I don't understand what you mean by 2. Why "replace true with false"? 


I meant lines 294, 334 and 340 in  e.g.

             window.addPreference(new PreferenceBoolean(Resources.getMessage("PropertyDialog.44"), true, true) { //$NON-NLS-1$
                protected Boolean getValue() { return model.getInputConfig().isHeuristicForSampleBasedCriteria(); }
                protected void setValue(Object t) { model.getInputConfig().setHeuristicForSampleBasedCriteria((Boolean)t); }});
            }

should be 

            window.addPreference(new PreferenceBoolean(Resources.getMessage("PropertyDialog.44"), false, true) { //$NON-NLS-1$
                protected Boolean getValue() { return model.getInputConfig().isHeuristicForSampleBasedCriteria(); }
                protected void setValue(Object t) { model.getInputConfig().setHeuristicForSampleBasedCriteria((Boolean)t); }});
            }


> I would suggest to not change line 49 but to add a new constructor public PreferenceBoolean(String label, Boolean hideLabel).

line 49 is modified to be:

        super(label,false);

I think this way the library reserves back-computability. I could not add a new constructor PreferenceBoolean(String label, Boolean hideLabel) as there is a constructor with the same signature i.e. 

 public PreferenceBoolean(String label, Boolean _default) 

> PS: Apart from that, I also think that it would be fine to keep the defaults in the ARX dialog. But they should be the correct defaults.

This is what I mean by "Correct the code in ARX"

Reply:
> line 49 is modified to be super(label,false);
> I think this way the library reserves back-computability.

No, this would break backwards compatibility (not "back-computability" ;)). The default is that if no default is provided, the default is "null". This would break this behaviour by setting the default to "false".

> I could not add a new constructor PreferenceBoolean(String label, Boolean hideLabel) as there is a constructor with the same signature i.e.

This is a really good point! So the solution would be to call
PreferenceBoolean(label, null, hideLabel) to achieve backward compatable behaviour.

-> No need to change swtpreferences.